framework/encode/capture_manager.h

```diff
@@ -306,24 +305,41 @@ class CaptureManager
         WriteToFile(scratch_buffer.data(), scratch_buffer.size());
     }
 
-  private:
-    static void AtExit()
+    void IncrementBlockIndex(uint64_t blocks)
     {
-        if (delete_instance_func_)
-        {
-            delete_instance_func_();
-            delete_instance_func_ = nullptr;
-        }
+        block_index_ += blocks;
+        GetThreadData()->block_index_ = block_index_;
     }
 
+  protected:
+    std::unique_ptr<util::Compressor> compressor_;
+    std::mutex                        mapped_memory_lock_;
+    util::Keyboard                    keyboard_;
+    std::string                       screenshot_prefix_;
+    util::ScreenshotFormat            screenshot_format_;
+    std::atomic<uint64_t>             block_index_;
+
+  private:
+    static void AtExit();
+
   private:
-    static uint32_t                                 instance_count_;
     static std::mutex                               instance_lock_;
+    static CommonCaptureManager*                    singleton_;
     static thread_local std::unique_ptr<ThreadData> thread_data_;
     static std::atomic<format::HandleId>            unique_id_counter_;
     static ApiCallMutexT                            api_call_mutex_;
 
-    const format::ApiFamilyId api_family_;
+    uint32_t instance_count_ = 0;
+    struct ApiInstanceRecord
+    {
+        size_t                count;
+        std::function<void()> destroyer;
+    };
+    using ApiCaptureManagerMap = std::unordered_map<ApiCaptureManager*, ApiInstanceRecord>;
+    ApiCaptureManagerMap api_capture_managers_;
+
+    CaptureSettings default_settings_;
```

Comment:
You should clarify the difference here between `default_settings_` and `capture_settings_` in a commet.